File: Challenges/Algorithms/Warmup/Staircase/staircase.py
#!/bin/python3

import sys


n = int(input().strip())

# Track Staircase State
num_spaces = n - 1;
num_hashes = 1;

# Iteratively Construct/Print Staircase
for row in range(n):

    # Print Spaces
    for i in range(num_spaces):
        print(" ", end="")
    num_spaces -= 1

    # Print Hashes
    for i in range(num_hashes):
        print("#", end="")
    num_hashes += 1

    # Advance to Next Row
    print("")

File: Challenges/Algorithms/Warmup/Staircase/staircase.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {

    // Track Staircase State
    unsigned int width = 0;
    std::cin >> width;
    unsigned int num_spaces = width - 1;
    unsigned int num_hashes = 1;

    // Iteratively Construct/Print Staircase
    for (unsigned int row = 0; row < width; row++) {
        // Print Spaces
        for (unsigned int i = 0; i < num_spaces; i++) {
            std::cout << " ";
        }
        num_spaces--;

        // Print Hashes
        for (unsigned int i = 0; i < num_hashes; i++) {
            std::cout << "#";
        }
        num_hashes++;

        // Advance to Next Row
        std::cout << std::endl;
    }

    return 0;
}
